Chris8205 patch 2

hylkevds opened the pull request:
Rebase of #1982 on v2.x

chris8205 committed:
Update values.yaml

Test Try

chris8205 committed:
common Labels Annotations

chris8205 committed:
Update http-ingress.yaml

add Version compatibility
add conditional to annotations
add conditional to rewriteroule

chris8205 committed:
Update values.yaml

chris8205 committed:
Update values.yaml

add same schema to mqtt ingress

chris8205 committed:
Update mqtt-ingress.yaml

add ingressClassName

chris8205 committed:
Update http-deployment.yaml

add lifeness and readyness

chris8205 committed:
Update mqtt-deployment.yaml

add liveness
readyness

chris8205 committed:
Update bus-deployment.yaml

lifeness probes

chris8205 committed:
Update db-deployment.yaml

lifeness probe

hamelchRIT committed:
Test

hamelchRIT committed:
many mods

hylkevds commented on helm/frost-server/templates/_tplvalues.tpl:
```diff
@@ -0,0 +1,38 @@
+{{/*
+Copyright Broadcom, Inc. All Rights Reserved.
```
We can't just copy code with a difference license into the FROST repo.

chris8205 replied:
can we fix it when we use 

dependencies:
  - name: common
    version: 2.22.0
    repository: oci://registry-1.docker.io/bitnamicharts

phertweck replied:
I think this would reference the helpers, developed [here](https://github.com/bitnami/charts/tree/main/bitnami/common). This might be a solution, but we need to discuss if we want to add the dependency only for two generic functions.

hylkevds replied:
Using a depencency is probably the cleanest and quickest solution. Could you make that change?

phertweck commented on helm/frost-server/templates/_helpers.tpl:
```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+*/}}
+{{- define "frost-server.http.serviceSubPath" -}}
+  {{- if not .Values.frost.http.urlSubPath | empty -}}
+      {{- printf "/%s/" .Values.frost.http.urlSubPath | replace "//" "/" -}}
```
Is this replacement needed?

phertweck commented on helm/frost-server/templates/_helpers.tpl:
```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+  {{- if not .Values.frost.mqtt.urlSubPath | empty -}}
+      {{- printf "/%s/" .Values.frost.mqtt.urlSubPath | replace "//" "/" -}}
+  {{- else -}}
+      {{- printf "/mqtt/" -}}
```
I'd suggest setting the default value in the `values.yaml`.

phertweck commented on helm/frost-server/templates/_helpers.tpl:
```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+  {{- if not .Values.frost.http.urlSubPath | empty -}}
+      {{- printf "/%s/" .Values.frost.http.urlSubPath | replace "//" "/" -}}
+  {{- else -}}
+      {{- printf "/" -}}
```
I'd suggest setting the default value in the `values.yaml`.

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -118,11 +118,15 @@ frost:
-      annotations:
+      enabled: true      
+      ## Set a specific ingress-class for the ingress ([default] not set)
+      className: nginx
```
I'd suggest to keep the ingress class empty by default to allow using the default ingress class of the cluster.

chris8205 replied:
yes default is "not set" maybe forgot the # while cleanup after testing...

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -118,11 +118,15 @@ frost:
+      ## Set a specific ingress-class for the ingress ([default] not set)
+      className: nginx
+      ## Defines the type of ingress-Controller in use (nginx [default], agic ) 
+      # ingressProvider: nginx 
```
What is this `ingressProvider` used for? Is it needed?

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -219,8 +223,26 @@ frost:
+        # cert-manager.io/cluster-issuer: letsencrypt
+        # cert-manager.io/issue-temporary-certificate: "true"
+      tls:
+        enabled: true
```
Default should be set to `false`, since the TLS-Secret is not part of the HELM-Chart.

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -219,8 +223,26 @@ frost:
+      ## Set a specific ingress-class for the ingress ([default] not set)
+      # className: azure-application-gateway
+      ## Defines the type of ingress-Controller in use (nginx [default], agic or traefik) 
+      # ingressProvider: nginx 
```
Same as above.

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -219,8 +223,26 @@ frost:
+    ingress:
+      enabled: false      
+      ## Set a specific ingress-class for the ingress ([default] not set)
+      # className: azure-application-gateway
```
I'd suggest not to comment out the className and leave it empty, by default to pic up the default ingress class

phertweck commented on helm/frost-server/values.yaml:
```diff
@@ -193,14 +197,14 @@ frost:
     enabled: true
     annotations: {}
-    serviceType: NodePort
+    serviceType: ClusterIP
```
I agree with this change, but since this is a breaking change we must be careful with the release.

phertweck commented on helm/frost-server/templates/_helpers.tpl:
```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+{{- end -}}
+
+{{/*
+Get the default agic rewriteAnnotations for HTTP.
```
Can this be combined wiht line 115?

phertweck commented on helm/frost-server/templates/_helpers.tpl:
```diff
@@ -33,20 +33,174 @@ Get the HTTP service API version.
+
+
+{{/*
+Get the default agic rewriteAnnotations for MQTT.
```
Shouldn't this be: "Get the default rewriteAnnotations for MQTT"? I'd suggest to move this over the specific annotation configurations.

phertweck commented on helm/frost-server/templates/db-deployment.yaml:
```diff
@@ -55,6 +55,36 @@ spec:
+                - /bin/sh
+                - -c
+                - exec pg_isready -h 127.0.0.1 -p 5432
+              # command:
```
I think this comment can be removed.

phertweck commented on helm/frost-server/templates/db-deployment.yaml:
```diff
@@ -55,6 +55,36 @@ spec:
+                - /bin/sh
+                - -c
+                - exec pg_isready -h 127.0.0.1 -p 5432
+              # command:
```
I think this comment can be removed.

phertweck commented on helm/frost-server/templates/http-ingress.yaml:
```diff
@@ -1,39 +1,52 @@
     component: {{ $tier }}
 spec:
+  {{- if and .Values.frost.http.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.frost.http.ingress.className | default "nginx" }}
```
I'd suggest to set the default value only in the `values.yaml`

phertweck commented on helm/frost-server/.vscode/settings.json:
This file shouldn't be in version control. Better add it to `.gitignore`.

phertweck reviewed (changes requested):
Thank you for the PR!

chris8205 committed:
multiple changes

chris8205 committed:
multiple changes2

chris8205 committed:
Delete helm/frost-server/.vscode directory

chris8205 committed:
Delete helm/frost-server/charts/common-2.22.0.tgz

mikedolx commented:
Hi @phertweck, 

do you need further information to process this PR, or are you waiting for @chris8205 to push the changes?

Thanks and BR,
Michael;

hylkevds commented:
> do you need further information to process this PR, or are you waiting for @chris8205 to push the changes?

There are some open questions of Philipp that are not addressed yet.
And the helm-build throws an error, that will need fixing too...

chris8205 committed:
fix templating issues

chris8205 commented:
> > do you need further information to process this PR, or are you waiting for @chris8205 to push the changes?
> 
> There are some open questions of Philipp that are not addressed yet. And the helm-build throws an error, that will need fixing too...

please take a look at 
chris8205:patch-1
i've worked a second time on it... but i think there is a mis match i dont know how i can push those changes into "patch2"

chris8205 committed:
fix accidently removed definition

hylkevds commented:
> please take a look at chris8205:patch-1 i've worked a second time on it... but i think there is a mis match i dont know how i can push those changes into "patch2"

I've cherry-picked your additional commits onto this branch, and the changes to the helm-subtree are now the same.
However, the helm-build is still failing: https://github.com/FraunhoferIOSB/FROST-Server/actions/runs/10954615154/job/30416872212?pr=1986

Also, can you document and breaking changes in the CHANGELOG.md file?

chris8205 committed:
fix helmbuild 'chris8205-patch-2'

/tmp/helm dependency update ./helm/frost-server

to get the dependency to the broadcom common-chart

hylkevds committed:
Updated helm version used for building
